src/ol/source/ogcTileUtil.js

```diff
@@ -96,14 +97,50 @@ const knownVectorMediaTypes = {
  * @property {Array<string>} [supportedMediaTypes] The supported media types.
  * @property {import("../proj/Projection.js").default} projection The source projection.
  * @property {Object} [context] Optional context for constructing the URL.
+ * @property {Array<string>} [collections] Optional collections to append the URL with.
  */
 
+/**
+ * @param {string} tileUrlTemplate Tile URL template.
+ * @param {Array<string>} collections List of collections to include as query parameter.
+ * @return {string} The tile URL template with appended collections query parameter.
+ */
+export function appendCollectionsQueryParam(tileUrlTemplate, collections) {
+  if (!collections.length) {
+    return tileUrlTemplate;
+  }
+
+  // making sure we can always construct a URL instance.
+  const url = new URL(tileUrlTemplate, 'file://');
```

Comment:
Minor but one slash is enough:

```suggestion
  const url = new URL(tileUrlTemplate, 'file:/');
```

Reply:
Oops. I forgot to wait for this. Maybe something for a quick follow-up fix.

Reply:
Created a follow up PR: https://github.com/openlayers/openlayers/pull/15809